plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -24,41 +26,104 @@ def manipulate_search_path(self, search_path: ConfigSearchPath) -> None:
         )
 
 
-def convert_to_deduced_type(string: str) -> Union[int, float, str]:
-    """Converts a string into a float or an int if it makes sense,
-    or returns the string.
+@dataclass
+class CommandlineSpec:
```

Comment:
This part could be moved somewhere else to share with other plugins

Reply:
cc @shagunsodhani

Reply:
Lets put it in a separate sub dir in the plugins dir (as discussed previously)

Reply:
I didn't get where that would be and how that would be imported that's why i had not so far, I can do it if you explain further though.
In the meantime I moved it to hydra/plugins/sweeper.py which seemed a good location

Reply:
This is not something that is relevant to all sweeper plugins, so I would rather keep it outside of the core.
it is also likely to evolve faster than the Hydra core as we start getting feedback about the nevergrad and ax plugins.

I propose:
plugins/common/hydra-hpo-core

which can be a regular pip package (not a plugin).
(hpo stands for hyper parameter optimization, open to other names).

Reply:
I do not have strong opinions but I find `hydra-opt-core` to more precise than `hydra-hpo-core`

Reply:
I won't add another installable package with this PR, I wouldnt have time to dig into all the little details that would need to be handled for that

Reply:
> I do not have strong opinions but I find hydra-opt-core to more precise than hydra-hpo-core

likestamp

> I won't add another installable package with this PR, I wouldnt have time to dig into all the little details that would need to be handled for that

it shouldn't be hard. you can base your setup.py it on one of the example plugins.
don't worry about actually publishing it. I will.